src/System.Management/tests/System/Management/ManagementObjectTests.cs

```diff
@@ -66,13 +65,24 @@ public void Invoke_Instance_And_Static_Method_Win32_Process()
             var processId = (uint)methodArgs[3];
             Assert.True(0u != processId, $"Unexpected process ID: {processId}");
 
-            var process = new ManagementObject($"Win32_Process.Handle=\"{processId}\"");
-            resultObj = process.InvokeMethod("Terminate", new object[]{ 0 });
-            resultCode = (uint)resultObj;
-            Assert.Equal(0u, resultCode);
+            // Before terminating the process open a handle to it so the processId cannot be re-used until
```

Comment:
Why not just store process.StartTime, and then if the PID still exists, check that it has changed?
This is how I solved this problem in a past codebase.
No need for pinvoke.

Reply:
That is also a good approach, but, this way I feel like we are ensuring (hopefully :smile:) that the test behaves consistently on every run instead of handling differences between runs. If this comes back I will add the StartTime check.